src/main/redux/sagas/auth.ts

```diff
@@ -453,35 +453,46 @@ async function opdsSetAuthCredentials(
     return [, new Error("")];
 }
 
-let _implicitNonce = "";
-const setAndGetImplicitNonceForImplicitAuthentication = () => (_implicitNonce = nanoid(16), _implicitNonce);
-const getImplicitNonceForImplicitAuthentication = () => _implicitNonce;
+const _implicitAuthData = { authenticationDocumentId: "", nonce: "" };
+const setAndGetImplicitNonceForImplicitAuthentication = () => (_implicitAuthData.nonce = nanoid(16), _implicitAuthData.nonce);
+const getImplicitAuthData = () => _implicitAuthData;
 
 function getHtmlAuthenticationUrl(auth: IOPDSAuthDocParsed) {
-
     let browserUrl: string;
     switch (auth.authenticationType) {
-
         case "http://opds-spec.org/auth/oauth/implicit": {
-
             try {
-                const browserUrlParsed = new URL(auth.links?.authenticate?.url);
-
-                // The unique identifier used in the callback URI must have the same value as the identifier provided in the Authentication Document.
-                // https://drafts.opds.io/authentication-for-opds-1.0#342-a-shared-client-identifier:~:text=The%20unique%20identifier%20used%20in%20the%20callback%20URI%20must%20have%20the%20same%20value%20as%20the%20identifier%20provided%20in%20the%20Authentication%20Document.
+                if (!auth.links?.authenticate?.url) {
+                    debug("OPDS Authentication Document Authentication Object property `authentication.links.href` null, empty, or missing.", auth.links?.authenticate?.url ?? "undefined");
+                    browserUrl = "";
+                    break;
+                }
 
-                // see also the callback response : it must use the id query parameter to indicate the Authentication Document identifier
-                // https://drafts.opds.io/authentication-for-opds-1.0#342-a-shared-client-identifier:~:text=it%20must%20use%20the%20id%20query%20parameter%20to%20indicate%20the%20Authentication%20Document%20identifier
+                const browserUrlParsed = new URL(auth.links?.authenticate?.url);
+                
+                // Record the OPDS Authentication Document's Id for later verification
+                const implicitAuthData = getImplicitAuthData();
+                implicitAuthData.authenticationDocumentId = auth.id;
 
-                browserUrlParsed.searchParams.set("client_id", encodeURIComponent_RFC3986(auth?.id || "http://opds-spec.org/auth/client"));
+                // client_id: Required. 
+                // Any Authentication Provider that supports OPDS Authentication 1.0 and exposes an Authentication Flow based on OAuth
+                // must identify all OPDS clients using the client_id "http://opds-spec.org/auth/client"
+                browserUrlParsed.searchParams.set("client_id", "http://opds-spec.org/auth/client");
```

Comment:
I don't know if it's good practice to hardcode the `client_id`. I understand that this can't be the same value as the authentication ID as I was able to code it in the first PR.
I think it's not a big deal and should work as there is only one "client" registered like Thorium.

What do you think @HadrienGardeur ? When you wrote the opds auth specification, did you think about this?
I poke @danielweck  also.

Reply:
Normally, an Authorization Server should have a unique `client_id` for every client app interacting with them.

That said, to allow public clients to interact with Authorization Servers, Authentication for OPDS introduces a shared client identifier along with a shared redirect URI. It's expected that OPDS clients will use this shared `client_id` as the default/fallback whenever they encounter an Authentication Document using OAuth. 

This is something that we'll have to revisit very soon though since we're adding support for Authorization Code with PKCE to a future revision of Authentication for OPDS.

Reply:
@panaC This response indicates that OPDS clients should use the shared `client_id` defined in the Authentication for OPDS spec. Is it ok to proceed like this?

Reply:
seems good for me